From 33fd4a60e8695d93813216a2c5c632de16fd2040 Mon Sep 17 00:00:00 2001
From: Luke Dashjr <luke-jr+git@utopios.org>
Date: Thu, 15 Dec 2011 18:00:36 -0500
Subject: [PATCH] kio_imap: fix APPEND without flags

d961033b4a5c02edbda8901a2c90349f859f6c00 accidentally changed the order of
precedence, making the message size (required) conditional on whether there
were any flags being set on the new message. This restores the original
precedence (while merging it with the new fast-concat rules), which should
make it IMAP4.1 compliant again.
---
 kioslave/imap4/imapcommand.cpp |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/kioslave/imap4/imapcommand.cpp b/kioslave/imap4/imapcommand.cpp
index 43daccd..18405a5 100644
--- a/kioslave/imap4/imapcommand.cpp
+++ b/kioslave/imap4/imapcommand.cpp
@@ -231,7 +231,11 @@ CommandPtr
 imapCommand::clientAppend (const QString & box, const QString & flags,
                            ulong size)
 {
-  QString tmp = QString(flags.isEmpty() ? QString("") : QString(('(' + flags + ") ")) + '{' + QString::number(size) + '}'); 
+  QString tmp = QString(
+    ((flags.isEmpty()) ?
+      QString("") :
+      QString(('(' + flags + ") "))) +
+    QString('{' + QString::number(size) + '}'));
   return CommandPtr( new imapCommand ("APPEND",
                           "\"" + KIMAP::encodeImapFolderName (box) + "\" " + tmp));
 }
-- 
1.7.3.4

